src/components/dashboard-sidebar.js

```diff
@@ -13,149 +12,152 @@ import MonetizationOnIcon from '@mui/icons-material/MonetizationOn';
   {
     href: '/xpub',
-    icon: (<AutoFixHighIcon fontSize="small" />),
-    title: 'xPub'
+    icon: <AutoFixHighIcon fontSize="small" />,
+    title: 'xPub',
   },
   {
     href: '/destination',
-    icon: (<LocationOnIcon fontSize="small" />),
-    title: 'Destination'
+    icon: <LocationOnIcon fontSize="small" />,
+    title: 'Destination',
   },
   {
     href: '/destinations',
-    icon: (<LocationSearchingIcon fontSize="small" />),
-    title: 'Destinations'
+    icon: <LocationSearchingIcon fontSize="small" />,
+    title: 'Destinations',
   },
   {
     href: '/destination-new',
-    icon: (<AddLocationIcon fontSize="small" />),
-    title: 'New Destination'
+    icon: <AddLocationIcon fontSize="small" />,
+    title: 'New Destination',
   },
   {
     href: '/transaction',
-    icon: (<MonetizationOnIcon fontSize="small" />),
-    title: 'Transaction'
+    icon: <MonetizationOnIcon fontSize="small" />,
+    title: 'Transaction',
   },
   {
     href: '/transactions',
-    icon: (<ViewListIcon fontSize="small" />),
-    title: 'Transactions'
+    icon: <ViewListIcon fontSize="small" />,
+    title: 'Transactions',
   },
   {
     href: '/transaction-new',
-    icon: (<AddIcon fontSize="small" />),
-    title: 'New Transaction'
+    icon: <AddIcon fontSize="small" />,
+    title: 'New Transaction',
   },
   {
     href: '/access-keys',
-    icon: (<KeyIcon fontSize="small" />),
-    title: 'Access Keys'
+    icon: <KeyIcon fontSize="small" />,
+    title: 'Access Keys',
   },
 ];
 
 export const DashboardSidebar = (props) => {
   const { open, onClose } = props;
-  const { xPubId, adminId, server } = useUser();
-  const { setAccessKeyString, setAdminKey, setXPrivString, setXPubString} = useModifyCredentials()
-  const navigate = useNavigate();
   const lgUp = useMediaQuery((theme) => theme.breakpoints.up('lg'), {
     defaultMatches: true,
-    noSsr: false
+    noSsr: false,
   });
 
-  const useItems = useMemo(() => {
-    if (adminId) {
-      if (xPubId) {
-        return [...adminItems, ...items];
+  const { type: credType, cred, clear: clearCredentials, server } = useCredentials();
+  const keyId = useMemo(() => {
+    if (credType === CredTypeNone) {
+      return '';
+    }
+    try {
+      if (credType === CredTypeAdmin) {
+        return bsv.crypto.Hash.sha256(Buffer.from(cred)).toString('hex');
       }
-      return [...adminItems];
-    } else {
-      return [...items];
+      // below calculate the hash for xPriv or accessKey cred
+      const keyObj =
+        credType === CredTypeXPriv
+          ? bsv.HDPrivateKey.fromString(cred).hdPublicKey
+          : bsv.PrivateKey.fromString(cred).publicKey;
+      return bsv.crypto.Hash.sha256(Buffer.from(keyObj.toString())).toString('hex');
+    } catch (e) {
+      console.error('Cannot calculate keyId', e);
+      return '';
     }
-  }, [adminId, xPubId]);
+  }, [cred, credType]);
+
+  const currentItems = credType === CredTypeAdmin ? adminItems : items;
```

Comment:
The `bsv` package is only used here. Maybe we will be able to skip this functionality and... thanks to that, don't have to implement `ts-sdk` package.

Reply:
What if instead of Id, we would just show 6 first and 6 last  chars of xpriv?
Something like:
`xprv9s****19CESK`

Reply:
Nice idea 👍 
However, I have to admit I haven't inspected the whole code good enough...It seems there are other places in this project where the `bsv` package is used. 

I think removing `bsv` entirely is out of scope for this PR, but we gave it a good shot!  😄

Reply:
Yes removing bsv is out of the scope, 
It should be rather in scope of replacing bsv with ts-sdk